tests/conftest.py

```diff
@@ -154,3 +161,176 @@ def stubbed_bursar_sfn_client():
+            logging.info(f"File s3://{bucket}/{key} already exists, nothing to do!")
+        except ClientError as e:
+            error_code = int(e.response["Error"]["Code"])
+            # ruff: noqa: PLR2004
+            if error_code == 404:
+                logging.info(f"File s3://{bucket}/{key} not found. Uploading...")
+                local_file_path = os.path.join("tests", "fixtures", os.path.basename(key))
+                if os.path.exists(local_file_path):
+                    s3.upload_file(local_file_path, bucket, key)
+                    logging.info(f"File uploaded to s3://{bucket}/{key}")
+                else:
+                    msg = f"Fixture file {local_file_path} does not exist."
+                    logging.exception(msg)
+                    raise FileNotFoundError(msg) from None
+            else:
+                raise
+
+    # Specify your bucket and key
+    fixtures = [
+        (
+            "dev-sftp-shared",
+            "exlibris/pod/POD_ALMA_EXPORT_20230815_220844[016]_new.tar.gz",
+        ),
+        (
+            "dev-sftp-shared",
+            "exlibris/timdex/TIMDEX_ALMA_EXPORT_DAILY_20230815_220844[016]_new.tar.gz",
+        ),
+    ]
+    for bucket, key in fixtures:
+        check_and_upload_file(bucket, key)
+
+
+@pytest.fixture
+def sample_webhook_post_body() -> dict:
+    return {
+        "action": "JOB_END",
+        "job_instance": {
+            "name": "Not a POD export job",
+        },
+    }
+
+
+@pytest.fixture
+def sample_pod_export_job_end_webhook_post_body() -> dict:
+    return {
+        "action": "JOB_END",
+        "job_instance": {
+            "name": os.getenv("ALMA_POD_EXPORT_JOB_NAME", "PPOD Export"),
+            "end_time": os.getenv("VALID_POD_EXPORT_DATE", "2022-05-23"),
+            "status": {"value": "COMPLETED_SUCCESS"},
+            "counter": [
+                {
+                    "type": {"value": "label.new.records", "desc": "New Records"},
+                    "value": "1",
+                },
+            ],
+        },
+    }
+
+
+@pytest.fixture
+def sample_timdex_export_job_end_webhook_post_body() -> dict:
+    return {
+        "action": "JOB_END",
+        "job_instance": {
+            "name": "Publishing Platform Job TIMDEX EXPORT to Dev1 DAILY",
+            "status": {"value": "COMPLETED_SUCCESS"},
+            "end_time": os.getenv("VALID_TIMDEX_EXPORT_DATE", "2022-10-24"),
+            "counter": [
+                {
+                    "type": {"value": "label.new.records", "desc": "New Records"},
+                    "value": "1",
+                },
+            ],
+        },
+    }
+
+
+def pytest_collection_modifyitems(config, items):
+    """Hook that is run after all tests collected, which allows for modification pre-run.
+
+    https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.hookspec.pytest_collection_modifyitems
+    """
+    # skip integration tests if WORKSPACE is not 'dev'
```

Comment:
Normally I'd agree, but this hook is a built-in function for pytests, and is called just by virtue of being named `pytest_collection_modifyitems`.  Therefore, if we want to add _other_ logic that should fire during this hook, it would also get added here.

But now that you mention it.. this behavior could be put into another function that this hook calls.  Then, that function's docstring can explain what it does, and future logic / functions could be added to this hook if needed.